pkg/spec/operation.go

```diff
@@ -312,24 +319,52 @@ func CloneOperation(op *spec.Operation) (*spec.Operation, error) {
 	return &out, nil
 }
 
+func handleAuthJWT(operation *spec.Operation, bearerToken string) (*spec.Operation, bool) {
+	// Parse the claims without validating (since we don't want to bother downloading a key)
+	parser := jwt.Parser{}
+	token, _, err := parser.ParseUnverified(bearerToken, jwt.MapClaims{})
+	if err != nil {
+		return operation, false
+	}
+	claims, ok := token.Claims.(jwt.MapClaims)
+	if !ok {
+		return operation, false
+	}
+	if scope, ok := claims["scope"]; ok {
+		scopes := strings.Split(scope.(string), " ")
+		log.Debugf("Found scopes: %v", scopes)
+		return addSecurity(operation, OAuth2SecurityDefinitionKey, scopes...), true
+	}
+	log.Warnf("No scope defined in this token")
+	return addSecurity(operation, OAuth2SecurityDefinitionKey, []string{}...), true
+}
+
 func handleAuthReqHeader(operation *spec.Operation, sd spec.SecurityDefinitions, value string) (*spec.Operation, spec.SecurityDefinitions) {
 	if strings.HasPrefix(value, BasicAuthPrefix) {
 		operation = addSecurity(operation, BasicAuthSecurityDefinitionKey)
 		sd = updateSecurityDefinitions(sd, BasicAuthSecurityDefinitionKey)
 	} else if strings.HasPrefix(value, BearerAuthPrefix) {
-		operation = addSecurity(operation, OAuth2SecurityDefinitionKey)
-		sd = updateSecurityDefinitions(sd, OAuth2SecurityDefinitionKey)
+		tokenString := strings.TrimPrefix(value, BearerAuthPrefix)
+		if len(tokenString) > 0 {
+			// Note: it might not be a JWT so we don't update in that case.
```

Comment:
Why not to update in this case?
https://oauth.net/2/access-tokens/
https://oauth.net/id-tokens-vs-access-tokens/
Looks like it can also be a random string string

Reply:
If it isn't a JWT, it will revert to the behavior of simply labeling it as OAuth with empty scopes as before. In that case, a random string like an API_KEY or the keys that Google uses would still be in the spec, but without any scopes.